test/mpi/test_mpi_utils.py

```diff
@@ -33,5 +34,14 @@ def test_key_split(_mpi_size, _mpi_comm):
 
     key, _ = nk.jax.mpi_split(key)
     keys = MPI.COMM_WORLD.allgather(key)
-    assert all([not jnp.all(k == keys) for k in keys])
```

Comment:
why this test was passing before?

Reply:
because it was correct.
but rng-split keys changed type in recent jax versions.

Reply:
I thought this was checking if the keys are different on different nodes, maybe I misunderstand the test, but it seems to me the old one was checking that after splitting the key is different, even on the same node

Reply:
it was only  checking that the key on a node is not equal to the original list of keys.

I just added a check that it also checks that the key on this node is equivalent to the key in the original array.